### PR TITLE
Revert: package folder for local install

### PR DESCRIPTION
**Pull request by lasote**
Reverted #10655

### CODE REVIEW

File: conans/client/conan_api.py [hunk 1]
```diff
@@ -601,8 +601,6 @@ def install(self, path="", name=None, version=None, user=None, channel=None,
                                         lockfile=lockfile)
 
             install_folder = _make_abs_path(install_folder, cwd)
-            if output_folder:
```
Comment (memsharded) [inline conans/client/conan_api.py]: Maybe this ``make_abs`` was still correct, when it is defined?

**Review by memsharded — COMMENTED** [review]
There were some changes in between. Like ``set_base_folders()`` now defines ``_base_package`` too for the local flow, which will be used by other commands like ``conan build``. Maybe that should be de-activated and tested too?

**Comment by SSE4** [comment]
what is the motivation to revert this fix? what is the alternative fix?

File: conans/client/conan_api.py (hunk 1 above)
Comment (lasote) [reply conans/client/conan_api.py]: Yes. I think this was probably a bug.

**Comment by lasote** [comment]
@SSE4 we consider that a local "conan install" should be done when you install your dependencies and you are developing locally your package (or project) so you should call the build system setting the prefix manually. What if I really want to build a project and install it in another folder or even the system? it is up to the user to call "cmake" or "autotools" setting the prefix. Also, it makes not much sense to set the base package folder to the output folder, where the build files will be, because even an editable package will take the "build" layout to locate the libraries. The 'cmake.install' and `make install` should be done in the `package()` method.

**Comment by SSE4** [comment]
@lasote thank you for the explanation, it makes sense.
however, can we improve an error message at least?
right now (1.45) I am getting:
```
ERROR: Traceback (most recent call last):
  File "/opt/pyenv/versions/3.7.5/lib/python3.7/site-packages/conans/errors.py", line 34, in conanfile_exception_formatter
    yield
  File "/opt/pyenv/versions/3.7.5/lib/python3.7/site-packages/conans/client/generators/__init__.py", line 245, in write_toolchain
    conanfile.generate()
  File "/host/bincrafters/conan-center-index/recipes/kmod/all/conanfile.py", line 62, in generate
    tc.generate()
  File "/opt/pyenv/versions/3.7.5/lib/python3.7/site-packages/conan/tools/gnu/autotoolstoolchain.py", line 152, in generate
    self.generate_args()
  File "/opt/pyenv/versions/3.7.5/lib/python3.7/site-packages/conan/tools/gnu/autotoolstoolchain.py", line 161, in generate_args
    ["--prefix=%s" % self._conanfile.package_folder.replace("\\", "/"),
AttributeError: 'NoneType' object has no attribute 'replace'
```
(it's with verbose traceback, otherwise it's just AttributeError)
which is not very helpful for the user

**Comment by lasote** [comment]
Sure, but that is more related to: https://github.com/conan-io/conan/pull/10672

**Review by franramirez688 — APPROVED** [review]

**Comment by memsharded** [comment]
Superseded by https://github.com/conan-io/conan/pull/10695

**Pull request closed without merging**